helpers/util.sh

```diff
@@ -264,8 +267,15 @@ buildmw() {
 
             pushd $PKG > /dev/null || die
             if [[ "$BUILDOFFLINE" = "" && "$PKG" != *"-localbuild" ]]; then
-                minfo "pulling updates..."
-                git pull >>$LOG 2>&1|| die "pulling of updates failed"
+                if [ -z "$GIT_REVISION" ]; then
+                    minfo "pulling updates..."
+                    git pull --recurse-submodules >>$LOG 2>&1|| die "pulling of updates failed"
+                else
+                    git diff-index --quiet HEAD -- || die "workdir of $PKG is not clean, refusing to reset version"
```

Comment:
Please be more explicit.
Unclean can be files from previous builds or uncommitted changes which is something else entirely,
I suggest:
```suggestion
                    git diff-index --quiet HEAD -- || die "Package $PKG has  is uncommitted changes, please commit first, refusing to reset version"
```

Reply:
git uses both synonymously:
nothing to commit, working tree clean

but both ways of writing it (seperately) are ok...

Reply:
@Thaodan opinions? I personally don't care what it says, I'd like to get this merged finally after over a year

Reply:
""Package $PKG has uncommitted changes, please commit first, refusing to reset version" sounds good, please adjust then it's good to go IMHO.

Reply:
Done